core/i18nilize/views.py

```diff
@@ -152,3 +152,204 @@ def patch(self, request):
+
+        if not is_valid_uuid(token_value):
+            return Response({'error': 'Invalid token.'}, status=status.HTTP_400_BAD_REQUEST)
+        
+        try:
+            token = Token.objects.get(value=token_value)
+        except Token.DoesNotExist:
+            return Response({'error': 'Missing valid token.'}, status=status.HTTP_404_NOT_FOUND)
+        
+        # get word and language to translate to from api body
+        original_word = request.query_params.get('original_word')
+        language = request.query_params.get('language')
+
+        # Check for missing parameters
+        if not original_word or not language:
+            return Response({"error": "Missing required fields in query params."}, status=status.HTTP_400_BAD_REQUEST)
+
+        # return translation if it exists
+        try:
+            translation = Translation.objects.get(token=token, original_word=original_word, language=language)
+            data = {
+                "language": translation.language,
+                "original_word": translation.original_word,
+                "translated_word": translation.translated_word
+            }
+            return Response(data, status=status.HTTP_200_OK)
+        
+        except Translation.DoesNotExist:
+            return Response({"error": "Translation not found for given language and word!"}, status=status.HTTP_404_NOT_FOUND)
+        
+    def patch(self, request):
+        """
+        Update a new single translation.
+        """
+        # get token if it exists
+        token_value = request.headers.get('Token')
+        if not token_value:
+            return Response({'error': 'Token is required.'}, status=status.HTTP_400_BAD_REQUEST)
+
+        if not is_valid_uuid(token_value):
+            return Response({'error': 'Invalid token.'}, status=status.HTTP_400_BAD_REQUEST)
+        
+        try:
+            token = Token.objects.get(value=token_value)
+        except Token.DoesNotExist:
+            return Response({'error': 'Missing valid token.'}, status=status.HTTP_404_NOT_FOUND)
+
+        # get translation from api body
+        if len(request.query_params) > 2:
+                return Response({"error": "query params should only include language and one translation pair!"}, status=status.HTTP_400_BAD_REQUEST)
+        language = request.query_params.get('language')
+        translation_pair = {key: value for key, value in request.query_params.items() if key != "language"}        
+
+        # validate query parameters
+        if not language or len(translation_pair.items()) < 1:
+            return Response({"error": "Missing required fields in query params."}, status=status.HTTP_400_BAD_REQUEST)
+        
+        original_word, translated_word = list(translation_pair.items())[0]
+        if original_word.isdigit() or translated_word.isdigit():
+            return Response({"error": "Translation pair must be in string format!"}, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
this code seems duplicated in the other 2 places as well, make this a helper function, something like `validate_translation_request` ? (you can come up with a better name if you dont like it)

Reply:
+1